core/block_svg.ts

```diff
@@ -235,29 +234,67 @@ export class BlockSvg
   }
 
   private computeAriaLabel(): string {
-    // Guess the block's aria label based on its field labels.
-    if (this.isShadow() || this.isSimpleReporter()) {
-      // TODO: Shadows may have more than one field.
-      // Shadow blocks are best represented directly by their field since they
-      // effectively operate like a field does for keyboard navigation purposes.
-      const field = Array.from(this.getFields())[0];
-      try {
-        return (
-          aria.getState(field.getFocusableElement(), aria.State.LABEL) ??
-          'Unknown?'
-        );
-      } catch {
-        return 'Unknown?';
+    const {blockSummary, inputCount} = buildBlockSummary(this);
+    const inputSummary = inputCount
+      ? ` ${inputCount} ${inputCount > 1 ? 'inputs' : 'input'}`
+      : '';
+
+    let currentBlock: Block | null = null;
+    let nestedStatementBlockCount = 0;
+    // This won't work well for if/else blocks.
+    this.inputList.forEach((input) => {
+      if (
+        input.connection &&
+        input.connection.type === ConnectionType.NEXT_STATEMENT
+      ) {
+        currentBlock = input.connection.targetBlock();
+      }
+    });
+    // The type is poorly inferred here.
+    while (currentBlock as Block | null) {
+      nestedStatementBlockCount++;
+      // The type is poorly inferred here.
+      // If currentBlock is null, we can't enter this while loop...
+      currentBlock = currentBlock!.getNextBlock();
+    }
+
+    let blockTypeText = 'block';
+    if (this.isShadow()) {
+      blockTypeText = 'input block';
+    } else if (this.outputConnection) {
+      blockTypeText = 'replacable block';
```

Comment:
I believe these two clauses are reversed; blocks with an output connection are input blocks, and shadow blocks get replaced when another block is dragged atop them.